### DISCUSSION TITLE
Wix 4.0 error WIX0200: The UI element contains an unhandled extension element 'WixUI'

### DISCUSSION THREAD

**Discussion by rnikey** [wixtoolset/issues#7592] [Questions] [answered]
I realise that there is a similar discussion recently, but this is WixUI.  I implemented the changes required from that discussion, being to add the extension via command "wix extension add -g WixToolset.Util.wixext" and then include the extension in the build command as is shown below.

I converted my v3 .wxs script to v4 using the conversion tool.

Declaration shows (as converted by the conversion tool)...

```xml
<Wix xmlns="http://wixtoolset.org/schemas/v4/wxs" xmlns:ui="http://wixtoolset.org/schemas/v4/wxs/ui">
```

Element reference further down (as converted by the conversion tool), now within the Package element, was Product element previously...

```xml
<ui:WixUI Id="WixUI_Advanced" />
```

Cut and paste from the command line...

```
C:\Users\Mikey\OneDrive\Development Battambang>wix build -o product.msi MikeyPersonal2.wxs -ext C:\Users\Mikey\.wix\extensions\WixToolset.Util.wixext\4.0.1\wixext4\WixToolset.Util.wixext.dll
C:\Users\Mikey\OneDrive\Development Battambang\MikeyPersonal2.wxs(224) : error WIX0200: The Package element contains an unhandled extension element 'WixUI'. Please ensure that the extension for elements in the 'http://wixtoolset.org/schemas/v4/wxs/ui' namespace has been provided.
```

What should I do?

**Comment by rnikey** [comment]
I did cut and paste the declaration, and it is shown if I attempt to edit the question.  I don't know why it's not showing above.

**Reply by robmen** [reply]
Fixed it for you. Fenced code blocks FTW.

**Reply by rnikey** [reply]
Thank you.  I'm retired now, just working on my own stuff and don't really want to learn Fenced code blocks.  If I were younger I would.  Maybe I still will if I have to keep posting, hopefully not though :).  Thanks for helping.

**Comment by robmen** [comment]
It turns out I covered exactly this topic very recently in the [Deployment Dojo](https://www.youtube.com/playlist?list=PLDlzbQXIs18slmqmdlS10_de_Cps-QRg6). Check out [Episode 37 - Using WixExtensions in WiX v4 to go beyond MSI functionality](https://youtube.com/live/-1-72Py0GSM?feature=share) for all the details.

**Reply by rnikey** [reply]
The details in my question should have alerted you to the fact that I probably had already watched that video.

**Comment by rnikey** [comment, marked as answer]
I found the problem.  I should have installed WixToolset.UI.wixext, as I meant to, and not WixToolset.Util.wixext.  Working too late and cutting and pasting the wrong line.
